erpnext/edi/doctype/code_list/code_list_import.py

```diff
@@ -0,0 +1,104 @@
+import json
+
+import frappe
+from frappe import _
+from frappe.utils.file_manager import save_file
+from lxml import etree
+
+
+@frappe.whitelist()
+def import_genericode():
+	doctype = frappe.form_dict.doctype
+	docname = frappe.form_dict.docname
+
+	# Parse the uploaded file content
+	parser = etree.XMLParser(remove_blank_text=True)
+	try:
+		root = etree.fromstring(frappe.local.uploaded_file, parser=parser)
+	except Exception as e:
+		frappe.throw(f"<pre>{e!s}</pre>", title=_("Parsing Error"))
+
+	# Extract the name (CanonicalVersionUri) from the parsed XML
+	name = root.find(".//CanonicalVersionUri").text
+
+	if docname:
+		code_list = frappe.get_doc(doctype, docname)
+		if code_list.name != name:
+			frappe.throw(_("The uploaded file does not match the selected Code List."))
+	else:
+		# Create a new Code List document with the extracted name
+		code_list = frappe.new_doc(doctype)
+		code_list.name = name
+		code_list.insert(ignore_permissions=True)
+
+	# Save the file using save_file utility
+	file_doc = save_file(
+		fname=frappe.local.uploaded_filename,
+		content=frappe.local.uploaded_file,
+		dt="Code List",
+		dn=code_list.name,
+		folder="Home/Attachments",
+		is_private=1,
+	)
+	file_path = file_doc.get_full_path()
+
+	# Get available columns and example values
+	columns, example_values, filterable_columns = get_columns_and_examples(file_path)
```

Comment:
```suggestion
	columns, example_values, filterable_columns = get_columns_and_examples(root)
```

I think we can pass the already parsed XML here.